src/Notepads/Controls/TextEditor/TextEditorCore.DateTime.cs

```diff
@@ -6,11 +6,36 @@
 
     public partial class TextEditorCore
     {
+        private bool _hasAddedLogEntry = false;
+
         private void InsertDateTimeString()
         {
             var dateStr = DateTime.Now.ToString(CultureInfo.CurrentCulture);
             Document.Selection.SetText(TextSetOptions.None, dateStr);
             Document.Selection.StartPosition = Document.Selection.EndPosition;
         }
+
+        /// <summary>
+        /// <para>
+        /// Adds "Windows Notepad" style header with current date and time, 
+        /// if a text document contains ".LOG" at the very beginning of file.
+        /// User can then add log entry from the very next line.
+        /// </para>
+        /// <see cref="https://support.microsoft.com/help/260563/how-to-use-notepad-to-create-a-log-file"/>
+        /// </summary>
+        public void TryInsertNewLogEntry()
+        {
+            var docText = _hasAddedLogEntry ? string.Empty : GetText();
+            if (string.IsNullOrEmpty(docText) || !docText.StartsWith(".LOG")) return;
```

Comment:
docText.StartsWith(".LOG")
What about a file looks like this:
-------
.LOG123456
7890
-------
Is this a valid .LOG file or no? Should we use .StartsWith(".LOG\r") here?

Reply:
> ## .LOG123456
> 7890
> Is this a valid .LOG file or no?

Yes, this is treated as valid .LOG file in default Notepad.